src/main/kotlin/me/zeroeightsix/kami/feature/module/Notebot.kt

```diff
@@ -0,0 +1,212 @@
+                ImGui.popStyleVar()
+            }
+        )
+    }
+
+    private fun discoverSong() {
+        mc.player?.let {
+            for (x in -4..4) {
+                for (y in -4..4) {
+                    for (z in -4..4) {
+                        val pos = it.pos?.plus(Vec3d(x.toDouble(), y.toDouble(), z.toDouble())) ?: break
+                        if (it.world.getBlockState(BlockPos(pos))?.block == Blocks.NOTE_BLOCK) {
+                            val noteBlock = it.world.getBlockState(BlockPos(pos))
+                            displayActionbar(it, "" + noteBlock.get(INSTRUMENT) + "[" + noteBlock.get(NOTE) + "]")
+                            instrumentMap.add(noteBlock.get(INSTRUMENT), noteBlock.get(NOTE), BlockPos(pos))
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    private fun loadSong() {
+        mc.player?.let {
+            playingSong = false
+            try {
+                MidiParser.parse("songs${File.separator}$songName").let {
+                    noteSequence = it.first
+                    channelMap = it.second
+                }
+                displayActionbar(it, "Loaded song ${File.separator}$songName")
+                lastNote = System.currentTimeMillis()
+                elapsed = 0
+            } catch (e: Exception) {
+                displayActionbar(it, "Sound not found ${File.separator}$songName")
+            }
+        }
+    }
+
+    @EventHandler
+    val listener = Listener<TickEvent.InGame>({
+        if (playingSong) {
+            val player = it.player
+            val world = it.world
+            if (!player.isCreative) {
+                while (noteSequence.isNotEmpty() && noteSequence.firstKey() <= elapsed) {
+                    playNotes(noteSequence.pollFirstEntry().value, player, world)
+                }
+                if (noteSequence.isEmpty()) playingSong = false
+                elapsed += System.currentTimeMillis() - lastNote
+            } else displayActionbar(player, "You are in creative mode and cannot play music.")
+        }
+        lastNote = System.currentTimeMillis()
+    })
+
+    private fun playNotes(notes: List<Note>, player: ClientPlayerEntity, world: World) {
+        val blockPosArr: ArrayList<BlockPos?> = ArrayList()
+        notes.forEach { n ->
+            val channelsArray: Array<Instrument> =
+                arrayOf(channelZero, channelOne, channelTwo, channelThree, channelFour)
+            val number = (n.track % (channelsArray.size - 1))
+            val enum = channelsArray[if (number < 0) 0 else number]
+            displayActionbar(player, enum.toString())
+            blockPosArr.add(instrumentMap[enum][if (enum == Instrument.HAT && hatAlwaysAsFSharp) 0 else n.notebotNote])
+        }
+        playBlock(blockPosArr, player, world)
+    }
+
+    private fun playBlock(blockPosTracks: ArrayList<BlockPos?>, player: ClientPlayerEntity, world: World) {
```

Comment:
Are you kidding me? You only broadened `world`, and forgot about broadening the list, for the second time!

```suggestion
    private fun playBlock(blockPosTracks: List<BlockPos?>, player: ClientPlayerEntity, world: World) {
```